src/main/kotlin/com/example/iot_management_device/exception/AuthenticationException.kt

```diff
@@ -0,0 +1,5 @@
+package com.example.iot_management_device.exception
+
+import org.springframework.security.authentication.BadCredentialsException
+
+class AuthenticationException(message: String, ex: BadCredentialsException) : RuntimeException("$message ${ex.message}")
```

Comment:
What the purpose to define this class?

Reply:
in the future we can expand this class adding a variety of authentication-related errors, not just those related to bad credentials

Reply:
By wrapping BadCredentialsException and other related exceptions (e.g., AccountLockedException, DisabledException) into AuthenticationException, we can handle all these cases in a uniform way and our @ControllerAdvice doesn’t need to have separate methods for each specific authentication-related exception.

so in future if we will expand this class it will be helpful